src/measure_theory/function/conditional_expectation.lean

```diff
@@ -2166,6 +2169,52 @@ begin
       (signed_measure.measurable_rn_deriv _ _).strongly_measurable },
 end
 
+/-- TODO: this should be generalized and proved using Jensen's inequality
+for the conditional expectation (not in mathlib yet) .-/
+lemma snorm_one_condexp_le_snorm {f : α → ℝ} :
+  snorm (μ[f | m]) 1 μ ≤ snorm f 1 μ :=
+begin
+  by_cases hf : integrable f μ,
+  swap,
+  { rw [snorm_congr_ae (condexp_undef hf), snorm_zero],
+    exact zero_le _ },
+  by_cases hm : m ≤ m0,
+  swap,
+  { rw [condexp_of_not_le hm, snorm_zero],
+    exact zero_le _ },
+  by_cases hsig : sigma_finite (μ.trim hm),
+  swap,
+  { rw [condexp_of_not_sigma_finite hm hsig, snorm_zero],
+    exact zero_le _ },
```

Comment:
```suggestion
  swap, { rw [snorm_congr_ae (condexp_undef hf), snorm_zero],  exact zero_le _ },
  by_cases hm : m ≤ m0,
  swap, { rw [condexp_of_not_le hm, snorm_zero], exact zero_le _ },
  by_cases hsig : sigma_finite (μ.trim hm),
  swap, { rw [condexp_of_not_sigma_finite hm hsig, snorm_zero], exact zero_le _ },
```
Does this fit into the character limit? If yes, then let's minimize the space taken by those unimportant degenerate cases.